.github/workflows/code-coverage-comment.yml

```
@@ -0,0 +1,17 @@
+name: Test and coverage
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+      - uses: actions/setup-go@v4
+        with: { go-version: '1.x' }
+      - name: Run coverage
+        run: go test -race -coverprofile=coverage.out -covermode=atomic
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
```

Can you try using the gen-coverage. It combines all Unit and Integration tests. Otherwise we only get the unit test coverage.
```suggestion
        run: make gen-coverage
      - name: Upload coverage to Codecov
        uses: codecov/codecov-action@v3
        with:
          files: ./tmp/total.cov
```

---

You could try adding the coverage to the `.github/workflows/test-integration.yml` workflow as it already starts opensearch but it would then run the test twice.

It might be needed to adjust the Makefile.
The `gen-coverage` executes the `go test` it self. Maybe it makes more sense to adjust the args of integ and unit test to match the current one of `gen-coverage`, so `gen-coverage` can just call the `make test-integ` and `make test-unit`.  Resulting in replacing the `make test-integ` in  `.github/workflows/test-integration.yml`  with `make gen-coverage`.
If this is to weird to understand, I can take a look tomorrow.

---

I have modified the makefile and the codecov action to make test-integ and make test-unit.